### DISCUSSION TITLE
URL Field in Search result webparts

### DISCUSSION THREAD

**Discussion by AnouckF** [microsoft-search/pnp-modern-search#3171] [Q&A] [answered]
Version used: 
Version 4.8

Question Description
I have set my search result webpart to search only within 1 specific SharePoint list. In this list, I have a field of type Hyperlink. Now I want the view to show this link. However, I have no way to retrieve it. When I put the webpart in Debug mode, I don't see this Link come back either.

Is there any way to still use this field in the search result webpart?

My List
![List](https://github.com/microsoft-search/pnp-modern-search/assets/17456461/1513de84-0a4f-40e6-bdf0-018f0aadbb70)

My search result webpar
![Schermafbeelding 2023-08-03 114753](https://github.com/microsoft-search/pnp-modern-search/assets/17456461/2daf2e61-302c-48a2-8274-b3257f2950a3)
t

**Comment by tnoatkins** [comment]
I'm following this post as I have the exact same issue! When I try to include the URL field as a search result, the result is always null.

**Reply by tnoatkins** [reply]
The URL I'm trying to read is stored in the WorkdayLink field in the original list, but using the Search Results debug mode, I can see that this field is returning null.
![image](https://github.com/microsoft-search/pnp-modern-search/assets/141351744/e47f48b5-136d-4e75-8752-eba220e9964d)

**Comment by kasperbolarsen** [comment]
Any chance you can upgrade to the current release version?
In general we can only provide any kind of assistance on the current release version.

**Comment by kasperbolarsen** [comment]
I'll see if I can recreate this issue in 4.9.1 and will revert later this week

**Reply by kasperbolarsen** [reply]
OK, that was a lot easier than expected. I can not recreate the issue as the hyperlink shows up just fine.

In my case the internal name of the site column is "searchHyperlink", and a managed property named "searchHyperlinkOSWURLH" is present ( I use the SP Search Query tool is verify). 
After having added the managed property to the Selected Properties list in the results web parts I can add the property to the columns and using HandleBars I can split the value in the URL and the alternative text components for display purposes.
In most cases people have forgotten to add the managed property to the selected properties list before added it to the display laoutout, could that be the reason in this case?

**Reply by tnoatkins** [reply]
Hey Kasper, can you think of a way to achieve this without setting the field as a managed property? I believe setting list columns to managed properties is locked for my non-admin account. Ideally, if there was a way to get all of the content of a list item, and then just parse the content for the url I'm looking for, that would be great, but I'm not sure if that exists.

**Reply by kasperbolarsen** [reply, marked as answer]
Hi, List Column? You did not create a Site Column for the Hyperlink field?
The managed property is automatically created for site columns, but I am not sure what happens when using List Columns.

**Reply by AnouckF** [reply]
I created a site column of my list column and now I'm able to get the information from the url field.

Is it possible to create an hyperlink of it?

**Reply by AnouckF** [reply]
I was able to fix it as the following 
<a href="{{getUrlField linkToSiteOWSURLH "URL"}}" target="_blank">
    {{getUrlField linkToSiteOWSURLH "Title"}}
</a>
